### ISSUE TITLE
Jason.DecodeError unexpected byte at position 0: 0x65 ("e")

### ISSUE THREAD

**Issue by eddy147** [restlessronin/openai_ex#89]
### bug description

When creating a run, by calling Runs.create/2, I get the following error on line 
https://github.com/restlessronin/openai_ex/blob/2f11e566bc520f92c8a0284b61ce275615c1e7b4/lib/openai_ex/http.ex#L86

In /lib/openai_ex/http.ex, function finch_run_no_decode, the finch_request looks like this:
```
|> Finch.request!(Map.get(openai, :finch_name), request_options(openai)) #=> %Finch.Response{
  status: 200,
  body: "event: thread.run.created\ndata: {\"id\":\"run_vGUg7qBnwXoOLk0DOEtrsBOT\",\"object\":\"thread.run\",\"created_at\":1715090122,\"assistant_id\":\"asst_xxx\",\"thread_id\":\"thread_xxx\",\"status\":\"queued\",\"started_at\":null,\"expires_at\":171
 
```
notice the body is not valid JSON, as it starts with "event: thread.run.created\ndata:"

The body is retrieved from the Map and used in Jason.decode! in finch_run right above it.

Now this goes wrong, because the body is not valid JSON.

I triggered the post by calling:

```elixir
  Runs.create(openai, %{
        thread_id: thread_id,
        assistant_id: Config.get_assistance_id(),
        stream: true
      })
   ```
   
 

### To Reproduce

call   

```elixir
Runs.create(openai, %{
        thread_id: thread_id,
        assistant_id: Config.get_assistance_id(),
        stream: true
      })
```

### Code snippets

See above

### OS

Linux (Pop_OS!)

### Elixir version

1.16.2

### Library version

openai_ex v0.6.0

**Comment by eddy147** [comment]
My bad.
I am running a local version of this repo, and added stream to the @api_fields of Runs. Hence the returned events.

**Comment by restlessronin** [comment]
@eddy147 Thanks for taking the time to report and triage the bug. ❤️❤️

**Comment by eddy147** [comment]
Shall I add a feature request to be able to use strem: true in Runs?

**Comment by restlessronin** [comment]
@eddy147 I have added support for run streams in [this branch](https://github.com/restlessronin/openai_ex/tree/run-stream).

This will work the same as the chat completions stream, i.e. it will return a map with the stream in the "body_stream" field.

Let me know if this works for you.

**Comment by restlessronin** [comment]
@eddy147 I have released v0.6.3 with support for run streaming.

**Comment by xantrac** [comment]
Hey @eddy147 thanks for the work on the stream. Unfortunately I'm still experiencing the same malformed payload on `v0.6.3`

```
    run_req =
      Runs.new(
        thread_id: thread_id,
        assistant_id: @assistant_id,
        model: @model_o
      )

    run_stream = openai() |> Runs.create(run_req, stream: true)

    run_stream.body_stream
    |> Stream.flat_map(& &1)
    |> Stream.each(&IO.inspect/1)
    |> Stream.run()
```

```
%{
  eventType: "thread.run.completed\ndata: {\"id\":\
```

**Comment by restlessronin** [comment]
@xantrac I opened a new issue to track this. Run stream weren't implemented until 0.6.3. Thanks for reporting ❤️

**Comment by xantrac** [comment]
> @xantrac I opened a [new issue](https://github.com/restlessronin/openai_ex/issues/92) to track this. Run stream weren't implemented until 0.6.3. Thanks for reporting ❤️

Awesome, thank you.

**Comment by restlessronin** [comment]
@xantrac looks like the run streaming format is not compliant with the SSE spec. I have asked about it on the [OpenAI forums](https://community.openai.com/t/assistants-streaming-not-conformant-with-sse-spec/760209)

In any case, I have applied a workaround that returns json with both 'event' and 'data' fields. Let me know if this fix works for you.

This has been published as v0.6.4.

Once again, thanks for bringing this to my attention.🙏🏾

**Comment by restlessronin** [comment]
> > @xantrac I opened a [new issue](https://github.com/restlessronin/openai_ex/issues/92) to track this. Run stream weren't implemented until 0.6.3. Thanks for reporting ❤️
> 
> Awesome, thank you.

@xantrac fixed in v0.6.4
